src/core/app.lua

```diff
@@ -70,16 +71,50 @@ end
 local function with_restart (app, method)
    if use_restart then
       -- Run fn in protected mode using pcall.
-      local status, err = pcall(method, app)
+      local status, result_or_error = pcall(method, app)
 
       -- If pcall caught an error mark app as "dead" (record time and cause
       -- of death).
-      if not status then app.dead = { error = err, time = now() } end
+      if not status then
+         app.dead = { error = result_or_error, time = now() }
+      end
+      return status, result_or_error
    else
-      method(app)
+      return true, method(app)
    end
 end
 
+-- Ingress packet drop monitor.
+ingress_drop_monitor = {
+   threshold = 100000,
+   wait = 20,
+   last_flush = 0,
+   last_value = ffi.new('uint64_t[1]'),
+   current_value = ffi.new('uint64_t[1]')
+}
+
+function ingress_drop_monitor:sample ()
+   local sum = self.current_value
+   sum[0] = 0
+   for i = 1, #app_array do
+      local app = app_array[i]
+      if app.ingress_packet_drops and not app.dead then
+         local status, value = with_restart(app, app.ingress_packet_drops)
+         if status then sum[0] = sum[0] + value end
+      end
+   end
+end
+
+function ingress_drop_monitor:jit_flush_if_needed ()
+   if self.current_value[0] - self.last_value[0] < self.threshold then return end
+   if now() - self.last_flush < self.wait then return end
+   self.last_flush = now()
+   self.last_value[0] = self.current_value[0]
+   jit.flush()
+   print("jit.flush")
```

Comment:
I am seeing my Snabb processes print `jit.flush` when running on `next` and as a user this is a little confusing. I think we should either make this message more verbose to explain to the user what is going on, and/or only print the message when debugging is enabled.

Example seeing the message during snabbnfv startup:

```
snabbnfv traffic starting (benchmark mode)
Loading program/snabbnfv/test_fixtures/nfvconfig/test_functions/snabbnfv-bench1.port
engine: start app B_NIC
engine: start app B_Virtio
jit.flush
Get features 0x28000
 VIRTIO_NET_F_CTRL_VQ VIRTIO_NET_F_MRG_RXBUF
Get features 0x28000
 VIRTIO_NET_F_CTRL_VQ VIRTIO_NET_F_MRG_RXBUF
load: time: 1.00s  fps: 0         fpGbps: 0.000 fpb: 0   bpp: -    sleep: 100 us
load: time: 1.00s  fps: 0         fpGbps: 0.000 fpb: 0   bpp: -    sleep: 100 us
```

and then later while processing traffic (that is always arriving faster than it can be processed, in this benchmark):
```
load: time: 1.00s  fps: 4,606,149 fpGbps: 2.690 fpb: 170 bpp: 64   sleep: 4   us
load: time: 1.00s  fps: 4,626,202 fpGbps: 2.702 fpb: 170 bpp: 64   sleep: 4   us
jit.flush
load: time: 1.00s  fps: 4,508,809 fpGbps: 2.633 fpb: 169 bpp: 64   sleep: 4   us
load: time: 1.00s  fps: 4,519,463 fpGbps: 2.639 fpb: 168 bpp: 64   sleep: 5   us
```